core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -248,7 +248,7 @@ object ConsoleConsumer extends Logging {
     val timeoutMsOpt = parser.accepts("timeout-ms", "If specified, exit if no message is available for consumption for the specified interval.")
       .withRequiredArg
       .describedAs("timeout_ms")
-      .ofType(classOf[java.lang.Integer])
+      .ofType(classOf[java.lang.Long])
```

Comment:
Why are we changing this?

Reply:
As I described above, it causes a type problem in Scala:

> As of present, `ConsoleConsumer` is taking timeout ms parameter as `Integer`. (see `ConsumerConfig#timeoutMsOpt`) For this reason, `ConsumerConfig#timeoutMs` is `Integer` and in turn, `timeoutMs` variable in `ConsoleConsumer#run` becomes `Any` - since it can either of `Integer` or `Long`.

Reply:
You could change the code in `run` to return `Int.MaxValue` instead of `Long.MaxValue`. That would not impact the command line parameters.

Reply:
Oh, I found much simple solution: call `toLong` in `ConsoleConsumer#run`.

```scala
val timeoutMs = if (conf.timeoutMs >= 0) conf.timeoutMs.toLong else Long.MaxValue
```